ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/worldstate/WorldStateArchive.java

```diff
@@ -28,20 +28,23 @@ public WorldStateArchive(final WorldStateStorage storage) {
     this.storage = storage;
   }
 
-  public WorldState get(final Hash rootHash) {
-    return getMutable(rootHash);
+  public Optional<WorldState> get(final Hash rootHash) {
+    return getMutable(rootHash).map(state -> state);
```

Comment:
Why is this `map` here?

Reply:
To make the compiler happy - it changes the type from `Optional<MutableWorldState>` to `Optional<WorldState>`.  Since `MutableWorldState` is an instance of `WorldState` you don't need to change the object but `Optional<MutableWorldState>` is not an instance of `Optional<WorldState>` so you do need to do a mapping.  

I'm sure someone can insert an appropriate rant about the limitations of Java generics here. :)